### PR TITLE
fix: Properly save reported targets

### PR DESCRIPTION
**Pull request by imurchie**
It turns out that on iOS 13 there is no relation between target id and page id (earlier page id `1` would be `page-1` as a target).

So handle mapping the three variables (app id, page id, target id).

Add Xcode 11/iOS 13 to Travis build.

### CODE REVIEW

File: lib/rpc-client.js [hunk 1]
```diff
@@ -238,6 +249,38 @@ export default class RpcClient {
...
+        data = JSON.parse(data);
+        const {targetInfo} = data.params;
+        this.addTarget(appIdKey, pageIdKey, targetInfo);
+        resolve();
+      });
+      await this.send('setSenderKey', {
+        appIdKey,
+        pageIdKey,
+      });
+      log.debug('Sender key set');
```
Comment (mykola-mokhnach) [inline lib/rpc-client.js]: should we print the key and page id itself

File: lib/rpc-client.js [hunk 2]
```diff
@@ -238,6 +249,38 @@ export default class RpcClient {
...
+      }
+    }).finally(() => {
+      this.deleteSpecialMessageHandler('_rpc_applicationSentData:');
+    });
+    this.shouldCheckForTarget = true;
+    await this.send('enablePage', {
+      appIdKey,
+      pageIdKey,
+    });
+    log.debug('Enabled activity on page');
```
Comment (mykola-mokhnach) [inline lib/rpc-client.js]: should we print the page id?

File: test/functional/safari-e2e-specs.js [hunk 3]
```diff
@@ -76,7 +77,9 @@ describe('Safari remote debugger', function () {
     await openUrl(sim.udid, address);
   });
   afterEach(async function () {
-    await rd.disconnect();
+    if (rd) {
+      await rd.disconnect();
```
Comment (mykola-mokhnach) [inline test/functional/safari-e2e-specs.js]: it probably makes sense to assign rd back to null after disconnect

File: lib/rpc-client.js [hunk 4]
```diff
@@ -238,6 +249,38 @@ export default class RpcClient {
...
 
+  async selectPage (appIdKey, pageIdKey) {
+    await new B(async (resolve, reject) => {
+      // this needs to be specially set here, rather than using the usual
+      // machinery, because the response is a generic data response, and it
+      // does not include the corresponding message id (i.e., there is no
+      // way to relate it back to the original message that was sent, and
+      // thus to the app and page)
+      this.setSpecialMessageHandler('_rpc_applicationSentData:', reject, (appId, data) => {
+        data = JSON.parse(data);
```
Comment (mykola-mokhnach) [inline lib/rpc-client.js]: can we expect an exception here?

File: lib/rpc-client.js (hunk 1 above)
Comment (imurchie) [reply lib/rpc-client.js]: They are printed before calling this function, and if a target is received that info is printed when it is processed.
```
dbug RemoteDebugger Selecting page '1' on app 'PID:38084' and forwarding socket setup
dbug RemoteDebugger Sending '_rpc_forwardSocketSetup:' message to remote debugger (id: 2)
dbug RemoteDebugger Sender key set
dbug RemoteDebugger Target created for app 'PID:38084' and page '1': {"targetId":"page-1","type":"page"}
```

File: lib/rpc-client.js (hunk 2 above)
Comment (imurchie) [reply lib/rpc-client.js]: See above.

File: lib/rpc-client.js (hunk 4 above)
Comment (imurchie) [reply lib/rpc-client.js]: `¯\_(ツ)_/¯`

It's Apple, so there's always the chance they messed up their own protocol.

File: lib/rpc-client.js [hunk 5]
```diff
@@ -257,7 +257,11 @@ export default class RpcClient {
       // way to relate it back to the original message that was sent, and
       // thus to the app and page)
       this.setSpecialMessageHandler('_rpc_applicationSentData:', reject, (appId, data) => {
-        data = JSON.parse(data);
+        try {
```
Comment (mykola-mokhnach) [inline lib/rpc-client.js]: this was always an interesting point for me whether such function is going to be rejected automatically if an exception happens

**Pull request merged**